app/models/graph.rb

```diff
@@ -1,5 +1,13 @@
+require 'timeout'
+
 module Graph
   class << self
+    def query(query_string, variables: {}, context: {}, timeout: nil)
+      Timeout.timeout(timeout) do
```

Comment:
Funny I had the same discussion with @cjoudrey yesterday.

Dylan showed me this https://github.com/ruby/ruby/blob/ruby_2_3/lib/net/http.rb#L878